Assets/Scripts/Interactable.cs

```diff
@@ -46,21 +116,146 @@ public class Interactable : MonoBehaviour {
+    if (refreshWithinRange) {
+      lastRefreshTime = currentTime;
+    }
+
+    foreach (var interactable in interactablesToCheck) {
+      if (interactable.shaderChanged) {
+        interactable.toggleOutlineShader();
+      }
+      bool interactorIsHoldingThisInteractable = itemHolder != null && itemHolder.HeldItem != null && itemHolder.HeldItem.Interactable == interactable;
+      if (!interactorIsHoldingThisInteractable) { // Skip an interactable if it's being held by the interactor or out of range
+        Vector3 interactablePosition = interactable.transform.position;
+        float distToLookingPoint = Vector3.Distance(lookingPoint, interactablePosition);
+        float distToCamera = Vector3.Distance(cameraPosition, interactablePosition);
+        bool inInteractionRange = distToCamera <= maxInteractionRange;
+        if (distToLookingPoint < closestInteractableDist && inInteractionRange) {
+          closestInteractableDist = distToLookingPoint;
+          closestInteractable = interactable;
+        }
+        if (inInteractionRange) {
+          withinRangeInteractables.Add(interactable);
+        } else {
+          withinRangeInteractables.Remove(interactable);
+        }
+      }
+    }
+
+    if (closestInteractable != null) {
+      Debug.DrawLine(lookingPoint, closestInteractable.transform.position, Color.cyan);
+    }
+
+    return closestInteractable;
+  }
+
+  public static void UseClosestInteractable(Vector3 interactorPosition, ItemHolder itemHolder) {
+    Interactable closestInteractable = Interactable.FindClosestInteractable(interactorPosition, itemHolder);
+    if (closestInteractable != null) {
+      closestInteractable.OnInteract(itemHolder);
+    }
+  }
+
+  public static void UseClosestInteractableInView(Vector3 cameraPosition, Vector3 cameraDirection, ItemHolder itemHolder) {
+    Interactable closestInteractable = Interactable.FindClosestInteractableInView(cameraPosition, cameraDirection, itemHolder);
+    if (closestInteractable != null) {
       closestInteractable.OnInteract(itemHolder);
     }
   }
+
+  public static void GiveClosestInteractableOutline(Vector3 interactorPosition, ItemHolder itemHolder) {
+    Interactable closestInteractable = Interactable.FindClosestInteractable(interactorPosition, itemHolder);
+    if (closestInteractable != null) {
+      if (!closestInteractable.shaderChanged) {
+        closestInteractable.toggleOutlineShader();
+      }
+      closestInteractable.showPrompt();
+    }
+  }
+
+  public static void GiveClosestInteractableInViewOutline(Vector3 cameraPosition, Vector3 cameraDirection, ItemHolder itemHolder) {
+    Interactable closestInteractable = Interactable.FindClosestInteractableInView(cameraPosition, cameraDirection, itemHolder);
+    if (closestInteractable != null) {
+      if (!closestInteractable.shaderChanged) {
+        closestInteractable.toggleOutlineShader();
+      }
+      closestInteractable.showPrompt();
+    }
+  }
+
+  public void showPrompt() {
```

Comment:
C# naming conventions suggest using PascalCase for all method names, I don't really care personally but that's just what the conventions say